### PR TITLE
EDA for oil consumption and renewable energy share

### PR DESCRIPTION
**Pull request by Tejas-Choudekar**

### CODE REVIEW

**Review by swapnalisonkusale — APPROVED** [review]

File: utils/preprocessing.py [hunk 1]
```diff
@@ -61,7 +61,27 @@ def preprocess_ocean_warming(df: pd.DataFrame) -> pd.DataFrame:
     df.rename(columns={'index': 'date', 'NOAA': 'ocean_warming'}, inplace=True)
     df['date'] = pd.to_datetime(df['date'])
     return df
+def preprocess_renewable_energy_share(df: pd.DataFrame) -> pd.DataFrame:
```
Comment (cristian-castro-a) [inline utils/preprocessing.py:66]: Please follow PEP8, and keep two empty lines between functions.

File: utils/preprocessing.py [hunk 2]
```diff
@@ -61,7 +61,27 @@ def preprocess_ocean_warming(df: pd.DataFrame) -> pd.DataFrame:
...
+    df = df[df['Entity'].str.contains( '(BP)' )==False]
+    df = df.groupby(by=['Year']).mean().reset_index()[['Year','Renewables (% equivalent primary energy)']]
+    df.insert(loc=1, column='month', value=12)
+    df.insert(loc=2, column='day', value=31)
+    df.insert(loc=0, column='date', value=pd.to_datetime(df[['Year','month','day']]))
+    df.drop(['Year','month','day'], axis=1, inplace=True)
+    df.rename(columns={'date': 'date', 'Renewables (% equivalent primary energy)': 'renewable_energy_share'}, inplace=True)
+    return df
 
+def preprocess_oil_consumption(df: pd.DataFrame) -> pd.DataFrame:
```
Comment (cristian-castro-a) [inline utils/preprocessing.py:78]: 2 empty lines between functions please.

**Review by cristian-castro-a — CHANGES_REQUESTED** [review]

File: utils/preprocessing.py (hunk 1 above)
Comment (Tejas-Choudekar) [reply utils/preprocessing.py:66]: Done

File: utils/preprocessing.py (hunk 2 above)
Comment (Tejas-Choudekar) [reply utils/preprocessing.py:78]: Done

**Review by cristian-castro-a — APPROVED** [review]

**Pull request merged**